parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -276,6 +276,10 @@ if [[ $OS == $UBUNTU_OS_NAME ]]; then
 fi
 {{end}}
 
+{{- if RunUnattendedUpgrades}}
+apt_get_update && unattended_upgrade
```

Comment:
My tests so far prove that the above works to ensure that when there are security updates available, running apt-get update and then running unattended-upgrade successfully, serially, gets them. So we can trust that the "runUnattendedUpgradesOnBootstrap" feature does the right thing and actually applies (i.e., reboots) the OS updates during cluster creation.

Reply:
So, in the past I saw this not always work but it could have been timing related to when other things are set up with respect to cloudinit.  This is likely a better place to do that.

Is there a reason that this would not be the default behavior?

Reply:
The primary reason is the judgment that having a node reboot before first coming online offers (1) undesirable delay and (2) demonstrable loss in node bootstrap reliability.

I don't think we can avoid #1, it's definitely going to take longer most of the time for nodes to come online if they come online with a stale OS security package configuration, and if they want to come up-to-date even if it requires a reboot. <-- is always going to drag up the average node bootstrap time

I wonder about #2 though. Can we summarize the additional risk of scooping up untested packages, plus any additional risk that a VM OS won't successfully come back online?

Reply:
The risk is relatively low but is not zero.  We have not had an outage due to the security updates as they are vetted relatively well.  The question is how bad is it to run a node without the security updates?

I am not saying someone could not opt out, but it is a question of which way we should be "safe by default" and what "safe" means.

Reply:
I claim we should start here and make a change to the default after some more testing maybe.

/lgtm